Sources/Plasma/PubUtilLib/plModifier/plDetectorLog.cpp

```diff
@@ -39,64 +39,27 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
       Mead, WA   99021
 
 *==LICENSE==*/
+
 #include "plDetectorLog.h"
-#include "plStatusLog/plStatusLog.h"
 
-// Don't bother logging detectors in the external release, since it isn't written to disk
 #ifdef PLASMA_EXTERNAL_RELEASE
-
-void DetectorLog(const char* format, ...) {}
-void DetectorLogSpecial(const char* format, ...) {}
-void DetectorLogRed(const char* format, ...) {}
-void DetectorLogYellow(const char* format, ...) {}
-
+plStatusLog* plDetectorLog::fLog = nullptr;
 #else
-
-static plStatusLog* gLog =
-plStatusLogMgr::GetInstance().CreateStatusLog(
-                                              20,
-                                              "Detector",
-                                              plStatusLog::kFilledBackground | plStatusLog::kDeleteForMe |
-                                              plStatusLog::kDontWriteFile | plStatusLog::kAlignToTop);
-
-
-void DetectorDoLogfile()
-{
-    delete gLog;
-    gLog = plStatusLogMgr::GetInstance().CreateStatusLog(20,"Detector.log",plStatusLog::kFilledBackground|plStatusLog::kDeleteForMe|plStatusLog::kAlignToTop);
-
-}
-
-void DetectorLog(const char* format, ...)
+plStatusLog* plDetectorLog::fLog = plStatusLogMgr::GetInstance().CreateStatusLog(
+    20,
+    "Detector.log",
+    plStatusLog::kFilledBackground | plStatusLog::kDeleteForMe |
+    plStatusLog::kDontWriteFile | plStatusLog::kAlignToTop);
+#endif
+
+void plDetectorLog::Output()
```

Comment:
I'm not sure the function name `Output` is any more meaningful than `DetectorDoLogfile` here...  What is the actual intent of the function?

Reply:
Generally, the detector log is not written to disk. This is called by a console command that causes the detector log to begin writing to disk.